graphdatascience/session/dedicated_sessions.py

```diff
@@ -74,12 +75,17 @@ def get_or_create(
         else:
             db_runner = self._create_db_runner(db_connection, neo4j_driver_options)
 
-            dbid = AuraApi.extract_id(db_connection.uri)
-            aura_db_instance = self._aura_api.list_instance(dbid)
+            aura_instance_id = AuraApi.extract_id(db_connection.uri) if not aura_instance_id else aura_instance_id
+            aura_db_instance = self._aura_api.list_instance(aura_instance_id)
 
             if aura_db_instance is None:
```

Comment:
could always have a deprecation warning if the aura_instance_id was not provided explicitly.
helping the user to migrate from the implicit behavior -- being future proof.